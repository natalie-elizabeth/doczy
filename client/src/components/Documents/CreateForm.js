import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import DocumentView from './documentView';

const CreateDocument = props => (
  <div>
    <TextField
      name="title"
      type="text"
      placeholder="Title"
      onChange={props.onTitleChange}
      defaultValue={props.document.title}
      fullWidth
    /> <br /><br />

    <TextField
      className="content"
      name="content"
      type="text"
      placeholder="Content"
      onChange={props.onContentChange}
      defaultValue={props.document.content}
      fullWidth
      multiLine
      rows={2}
      rowsMax={10}
    /><br /> <br />

    <SelectField
      floatingLabelText="Access"
      value={props.document.access}
      onChange={props.onSetAccess}
    >
      <MenuItem value={'public'} primaryText="Public" />
      <MenuItem value={'private'} primaryText="Private" />
      <MenuItem value={'role'} primaryText="Role" />
    </SelectField>
  </div>
);

export default CreateDocument;
