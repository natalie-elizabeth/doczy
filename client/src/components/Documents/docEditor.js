import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import DocumentForm from './documentForm';



class componentName extends Component {
  render() {
    return (
      <div>

      </div>
    );
  }
}

export default componentName;
