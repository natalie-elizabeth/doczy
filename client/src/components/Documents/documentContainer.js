
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import Snackbar from 'material-ui/Snackbar';
import * as documentActions from '../../actions/docActions';
import DocumentView from './documentView';
import DocumentList from './documentList';
import CreateDocument from './CreateForm';
import DocumentEditForm from './editDocument';
import Pagination from 'react-js-pagination';
import DocumentSearch from './documentSearch';

import * as tokenUtils from '../../utils/tokenUtils';

const style = {
  position: 'fixed',
  top: 70,
  right: 20,
  marginRight: 30,
};

export class DocumentViewContainer extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      open: false,
      activePage: 1,
      limit: 3,
      offset: 0,
      document: {
        title: '',
        content: '',
        access: '',
        snackBarOpen: false

      }
    };
    this.handleOpen = this.handleOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.onSetAccess = this.onSetAccess.bind(this);
    this.onTitleChange = this.onTitleChange.bind(this);
    this.onContentChange = this.onContentChange.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handlePageChange = this.handlePageChange.bind(this);
    this.closeSnackBar = this.closeSnackBar.bind(this);
  }
  componentWillMount() {
    this.props.documentActions.listDocuments(this.state.limit, this.state.offset);
  }
  onSetAccess(event, index, value) {
    const Document = this.state.document;
    Document.access = value;
    this.setState({ document: Document });
  }
  onTitleChange(event) {
    const Document = this.state.document;
    Document.title = event.target.value;
    this.setState({ document: Document });
  }
  onContentChange(event) {
    const Document = this.state.document;
    Document.content = event.target.value;
    this.setState({ document: Document });
  }

  handleOpen() {
    this.setState({ open: true, document: {} });
  }

  handleClose() {
    this.setState({ open: false, edit: false });
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState({
      document: Object.assign({}, this.state.document, {
        [name]: value
      })
    });
  }
  closeSnackBar() {
    this.setState({ snackBarOpen: false });
  }

  handlePageChange(pageNumber) {
    this.setState({ activePage: pageNumber });
    this.props.documentActions.listDocuments(this.state.limit, (this.state.limit * (pageNumber - 1)));
  }

  updateDocument(document) {
    return event => {
      this.setState({
        document,
        edit: true,
        open: true,
      });
    };
  }

  handleSubmit(event) {
    event.preventDefault();
    this.state.edit
      ? this.props.documentActions.updateDocument(this.state.document)
      : this.props.documentActions.createDocument(this.state.document);
    this.setState({ snackBarOpen: true });
    this.handleClose();

  }

  render() {
    const viewActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={this.handleClose}
        className="waves-effect waves-light btn"
      />,
      <FlatButton
        label="Submit"
        primary
        keyboardFocused
        onTouchTap={this.handleSubmit}
      />,
    ];
    return (
      < div className="container" >

        <div>
          <div style={{ marginLeft: '30px' }}> <DocumentSearch /> </div>
          <hr />
          {this.props.documentList.documents && this.props.documentList.documents.length ? (this.props.documentList.documents.map(document =>
            (<DocumentView
              key={document.id}
              document={document}
              onUpdate={this.updateDocument(document)}
              deleteDocument={this.props.documentActions.deleteDocument}
              listDocuments={this.props.documentActions.listDocuments}


            />)
          )) : <p style={{ fontSize: "20px" }}>No Documents Available</p>}
          <div className="paginitaion" >
            <Pagination
              activePage={this.state.activePage}
              itemsCountPerPage={this.state.limit}
              totalItemsCount={10}
              pageRangeDisplayed={3}
              onChange={this.handlePageChange}
            />
          </div>

          <div>
            <FloatingActionButton onClick={this.handleOpen} backgroundColor="#681140" style={style}>
              <ContentAdd />
            </FloatingActionButton>
          </div>
        </div>
        <br />
        <Dialog
          title="Create a new Document"
          actions={viewActions}
          modal={false}
          open={this.state.open}
          onRequestClose={this.handleClose}
        >
          {this.state.edit ? (
            <DocumentEditForm
              document={this.state.document}
              onChange={this.handleChange}
            />
          ) : (
              <CreateDocument
                style={style}
                onSetAccess={this.onSetAccess}
                document={this.state.document}
                onTitleChange={this.onTitleChange}
                onContentChange={this.onContentChange}
              />
            )}
        </Dialog>
        <Snackbar
          open={this.state.snackBarOpen}
          message="Document Saved"
          autoHideDuration={2000}
          onRequestClose={this.closeSnackBar}
        />
    </div >

    );
  }
}
DocumentViewContainer.propTypes = {
  documentList: PropTypes.object.isRequired,
  deleteDocument: PropTypes.func.isRequired,
  documentActions: PropTypes.object.isRequired,
  router: PropTypes.object.isRequired
};

DocumentViewContainer.contextTypes = {
  router: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    documentList: state.documents
  };
}

function mapDispatchToProps(dispatch) {
  return {
    documentActions: bindActionCreators(documentActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(DocumentViewContainer);
