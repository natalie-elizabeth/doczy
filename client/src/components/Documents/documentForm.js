import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';

import MenuItem from 'material-ui/MenuItem';
import SelectField from 'material-ui/SelectField';
import DocumentView from './documentView';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Card, CardText } from 'material-ui/Card';

const DocumentForm = props => (
  <div>
    <MuiThemeProvider>
      <Card >
        <TextField
          type="text"
          name="title"
          placeholder="Document Title"
          defaultValue={props.document.title}
          onChange={props.onTitleChange}
          fullWidth
        />
        <br />
        <TextField
          type="text"
          name="content"
          placeholder="Write your document"
          defaultValue={props.document.content}
          onChange={props.onContentChange}
          multiLine
          fullWidth
        />
        <br />
        <SelectField
          floatingLabelText="Access"
          value={props.document.access}
          onChange={props.onSetAccess}
        >
          <MenuItem value={'public'} primaryText="Public" />
          <MenuItem value={'private'} primaryText="Private" />

        </SelectField>
      </Card>
    </MuiThemeProvider>
  </div>
);


export default DocumentForm;
