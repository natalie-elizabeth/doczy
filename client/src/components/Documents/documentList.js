import React from 'react';
import PropTypes from 'prop-types';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import DocumentView from './documentView';
import { Card, CardText } from 'material-ui/Card';

const DocumentList = ({ documents, handleOpen, style }) => (
  <div>
    <Card>
      {documents.map(document =>
        <DocumentView key={document.id} document={document} />
      )}
    </Card>
    <div>
      <FloatingActionButton onClick={handleOpen} style={style}>
        <ContentAdd />
      </FloatingActionButton>
    </div>
  </div >
);
DocumentList.propTypes = {
  documents: PropTypes.array.isRequired,
  handleOpen: PropTypes.func.isRequired,
  style: PropTypes.func.isRequired
};

export default DocumentList;
