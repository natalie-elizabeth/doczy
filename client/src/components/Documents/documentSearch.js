import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import _ from 'lodash';
import * as docActions from '../../actions/docActions';


export class DocumentSearch extends React.Component {
  constructor(props, context) {
    super(props);
    this.searchDocument = this.searchDocument.bind(this);
    this.handleSearchInput = this.handleSearchInput.bind(this);
    this.apiCall = this.apiCall.bind(this);
    this.makeSearch = _.debounce(this.apiCall, 500);
  }
  searchDocument(searchFilter) {
    this.props.searchDocument(searchFilter);
  }

  apiCall() {
    // console.log('>>>>>>>>>>>>>>>>', this.state.searchFilter);
    this.props.searchDocument(this.state.searchFilter);
  }
  handleSearchInput(event) {
    this.setState({ searchFilter: event.target.value });
    this.makeSearch();
  }


  render() {
    return (
      <div className="search-wrapper card" style={{ marginLeft: '80%' }}>

        <i className="material-icons" style={{ color: "black" }}>search</i>
        <input
          id="search"
          onChange={this.handleSearchInput}
        />

      </div>

    );
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(docActions, dispatch);
}

export default connect(null, mapDispatchToProps)(DocumentSearch);
