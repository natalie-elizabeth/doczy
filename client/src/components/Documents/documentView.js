import React, { PropTypes } from 'react';
import Chip from 'material-ui/Chip';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import { GridList, GridTile } from 'material-ui/GridList';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import JWTdecode from 'jwt-decode';
import StarBorder from 'material-ui/svg-icons/toggle/star-border';
import Header from '../common/Header';
import { getUserFromToken } from '../../utils/tokenUtils';

const styles = {
  titleStyle: {
    color: '#ffffff',
    font: 'cursive'
  }
};

const DocumentView = props => (
  <div>
    <GridList
      cellHeight="auto"
      cols={1}
    >
      {getUserFromToken().userId === props.document.userId ?
        <IconMenu
          style={{ float: 'right' }}
          iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
          anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
          targetOrigin={{ horizontal: 'right', vertical: 'top' }}
        >
          <MenuItem
            primaryText="Edit Document"
            onTouchTap={props.onUpdate}
          />
          <MenuItem
            primaryText="Delete Document"
            onTouchTap={() => {
              props.deleteDocument(props.document.id)
                .then(() => {
                  props.listDocuments();
                });
            }
            }
          />
        </IconMenu> : <span />
      }
      <GridTile>
        <h2>{props.document.title}</h2>
        {props.document.content}
        <Chip backgroundColor="#123c69" labelColor="#ffffff"> {props.document.access} </Chip> <br />
      </GridTile>
    </GridList>
  </div>
);

Document.propTypes = {
  dispatch: PropTypes.func.isRequired,
  documents: PropTypes.func.isRequired,
  onUpdate: PropTypes.func.isRequired,
};

export default DocumentView;
