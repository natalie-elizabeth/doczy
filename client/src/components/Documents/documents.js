import React, { Component } from 'react';
import DocumentForm from './documentForm';

export default class Home extends Component {
  render() {
    return (
      <div>
        <DocumentForm />
      </div>
    );
  }
}
