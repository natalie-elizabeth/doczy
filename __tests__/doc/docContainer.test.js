import React from 'react';
import shallowToJSON from 'enzyme-to-json';
import { mountWithContext as mount, shallowWithContext as shallow } from '../test-utils';
import { DocumentViewContainer } from '../../client/src/components/Documents/documentContainer';


const constructWrapper = props => shallow(<DocumentViewContainer {...props} />);
const constructProps = ({ documents = [] } = {}) => ({
  deleteDocument() { },
  router: {},
  documentActions: {
    listDocuments() { }
  },
  documentList: {
    documents
  },
});

describe('document container', () => {
  it('should render correctly', () => {
    const wrapper = constructWrapper(constructProps());
    expect(wrapper.props().className).toBe('container');
  });
  it('should have a matching snapshot', () => {
    const wrapper = constructWrapper(constructProps());
    const tree = shallowToJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have a dialog container', () => {
    const wrapper = constructWrapper(constructProps());
    expect(wrapper.find('Dialog').length).toEqual(1);
  });
  it('should handleClose', () => {
    beforeEach((done) => {
      setTimeout(() => {
        done();
      }, 500);
    });
  });
  it('handleOpen opens the modal', () => {
    const props = constructProps();
    const wrapper = constructWrapper(props);

    let state = wrapper.state();
    expect(state.open).toBe(false);
    expect(wrapper.find('Dialog').props().open).toBe(false);

    const button = wrapper.find('FloatingActionButton').first();
    button.simulate('click');

    state = wrapper.state();
    expect(state.open).toBe(true);
    expect(wrapper.find('Dialog').props().open).toBe(true);
  });

  it('handleClose closes the modal', () => {
    const props = constructProps();
    const wrapper = constructWrapper(props);

    let state = wrapper.state();
    expect(state.open).toBe(false);
    expect(wrapper.find('Dialog').props().open).toBe(false);

    const button = wrapper.find('FlatButton').first();
    button.simulate('click');

    state = wrapper.state();
    expect(state.open).toBe(false);
    expect(wrapper.find('Dialog').props().open).toBe(false);
  });
});
