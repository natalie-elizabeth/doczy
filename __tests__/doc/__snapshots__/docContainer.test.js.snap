// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document container should have a matching snapshot 1`] = `undefined`;
